# HG changeset patch
# User Rik <rdrider0-list@yahoo.com>
# Date 1259092621 28800
#      Tue Nov 24 11:57:01 2009 -0800
# Node ID cd53ecf0d79a6e36626ae25d0d683184dd77b2c9
# Parent  c0d0b6e37a36ceb03123b6d85a34993853e9c385
Correct Makefile.am so that make will halt on error when unable to build doc-cache

Subtle error where shell command A || CLEANUP was returning the exit status of
CLEANUP on an error rather than an error code because command A failed.

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,13 @@
+2009-11-24  Rik <octave@nomad.inbox5.com>
+
+	* interpreter/Makefile.am: 
+	Correct Makefile so that it halts on error when unable to build doc-cache
+
 2009-11-23  Rik <octave@nomad.inbox5.com>
 
 	* faq/Makefile.am, interpreter/Makefile.am, liboctave/Makefile.am: 
 	Stop distribution of conf.texi by autotools.  conf.texi must be re-made
 	from conf.texi.in during configure step
 
 2009-11-23  Rik <octave@nomad.inbox5.com>
 
diff --git a/doc/interpreter/Makefile.am b/doc/interpreter/Makefile.am
--- a/doc/interpreter/Makefile.am
+++ b/doc/interpreter/Makefile.am
@@ -199,17 +199,17 @@ DOCSTRING_FILES = $(TOPDIR)/src/DOCSTRIN
 
 $(TOPDIR)/src/DOCSTRINGS:
 	$(MAKE) -C $(TOPDIR)/src DOCSTRINGS
 
 $(TOPDIR)/scripts/DOCSTRINGS:
 	$(MAKE) -C $(TOPDIR)/scripts DOCSTRINGS
 
 doc-cache: $(DOCSTRING_FILES) mk_doc_cache.m
-	$(TOPDIR)/run-octave -f -q -H $(srcdir)/mk_doc_cache.m doc-cache $(DOCSTRING_FILES) || rm -f doc-cache
+	$(TOPDIR)/run-octave -f -q -H $(srcdir)/mk_doc_cache.m doc-cache $(DOCSTRING_FILES) || { rm -f doc-cache; exit 1; }
 
 $(MUNGED_TEXI_SRC): $(DOCSTRING_FILES) munge-texi$(BUILD_EXEEXT)
 
 munge-texi$(BUILD_EXEEXT): munge-texi.cc
 	$(BUILD_CXX) $(BUILD_CXXFLAGS) -o $@ $^ $(BUILD_LDFLAGS)
 
 contributors.texi: contributors.in
 	$(AWK) -f $(srcdir)/mkcontrib.awk $(srcdir)/contributors.in > $@-t
